easybuild/easyblocks/l/lammps.py

```diff
@@ -119,6 +121,17 @@ def prepare_step(self, *args, **kwargs):
         if self.cfg['kokkos'] and get_software_root('CUDA'):
             env.unset_env_vars(['LIBS'])
 
+        # Add files from Atomistica
+        atomistica = get_software_root('Atomistica')
```

Comment:
Shouldn't this look at the dependencies? I think it was `self.dependencies()`

Reply:
@Flamefire You are a bit of a CMake guru, is there a way to trigger linking against the Fortran MPI libraries? The think is I believe it should already be there  from https://github.com/lammps/lammps/blob/master/cmake/CMakeLists.txt#L706 so I wonder is CMake not correctly detecting MPI?

Reply:
It's hard. Let me give you some information on what's happening (If I'm not mistaken and they e.g. use a custom FindMPI in CMake [e.g. https://github.com/lammps/lammps/blob/b38e95f88c001b6cad1097f0e8a4d65508582ca8/cmake/CMakeLists.txt#L131]):

- They search for MPI https://github.com/lammps/lammps/blob/b38e95f88c001b6cad1097f0e8a4d65508582ca8/cmake/CMakeLists.txt#L155
- This creates imported targets containing the MPI libs, IIRC this is done by invoking the mpi wrapper and querying the libraries to use.
- This imported library is then linked to lammps: https://github.com/lammps/lammps/blob/b38e95f88c001b6cad1097f0e8a4d65508582ca8/cmake/CMakeLists.txt#L156
- Some information about found libraries is given as a message only: https://github.com/lammps/lammps/blob/b38e95f88c001b6cad1097f0e8a4d65508582ca8/cmake/CMakeLists.txt#L707-L709

Note how they explicitly link against `MPI::MPI_CXX`, so no fortran there.

Question I have: If they use MPI from C++ and Atomistica uses MPI from Fortran, don't you need to link against the MPI C++ and the MPI Fortran lib?

If so, then it would be best to write a simple CMake find script:

```
find_library(ATOMISITICA_LIB atomistica)
find_package(MPI COMPONENTS Fortran)
add_library(Atomistica::Atomistica STATIC IMPORTED)
set_target_properties(Atomistica::Atomistica
  IMPORTED LOCATION ${ATOMISITICA_LIB}
  INTERFACE_LINK_LIBRARIES MPI::MPI_Fortran
  IMPORTED_LINK_INTERFACE_LANGUAGES Fortran
)
include(FindPackageHandleStandardArgs)
find_package_handle_standard_args(Atomistica REQUIRED_VARS ATOMISITICA_LIB MPI_Fortran_FOUND)
```

The created library can then be linked against the lammps library:
- Save above to `FindAtomistica.cmake` in cmake/Modules of lammps
- Add call to `find_package(Atomistica REQUIRED)` to the CML
- Add `target_link_libraries(lammps PUBLIC Atomistica::Atomistica)`

If that works, the `FindAtomistica.cmake` can be contributed to LAMMPS so they can properly do it and make our job in the future easier, see https://github.com/lammps/lammps/issues/2039

Reply:
@Flamefire Thanks for going through the cmake build of LAMMPS and pointing out the required changes to handle a build linking with Atomistica. I've adapted them into a patch that works perfectly well :+1:

Reply:
Patch is in https://github.com/easybuilders/easybuild-easyconfigs/pull/10433/commits/72b8adeb21dcc8dc10ccd5eed27737197f5a3455